### DISCUSSION TITLE
Does pgx implement a way to prevent SQL Injections?

### DISCUSSION THREAD

**Discussion by DanielCoder834** [jackc/pgx#2099] [Q&A] [answered]
I found this [documentation](https://go.dev/doc/database/sql-injection) on how the sql library prevents SQL injection. Does the pgx have something similar to the method they introduce? Thank you for any help on the matter.

**Comment by jackc** [comment, marked as answer]
Yes, if you are using $1, $2, etc. then SQL injection is prevented. pgx uses the PostgreSQL extended protocol / prepared statements by default where the SQL string and the arguments are passed separately to the PostgreSQL database. When using the simple protocol, pgx handles variable sanitization and interpolation.

**Reply by DanielCoder834** [reply]
Awesome thank you so much for the reply. Does that also apply to named arguments as well?

**Reply by jackc** [reply]
Named arguments? If you mean pgx's https://pkg.go.dev/github.com/jackc/pgx/v5#NamedArgs then yes. Otherwise, PostgreSQL doesn't support named arguments directly. Everything is $1, $2, etc. If whatever you are using eventually converts to $1 style then it is safe.

**Reply by DanielCoder834** [reply]
Yes I meant that. Thank you
